src/status_im/communities/core.cljs

```diff
@@ -87,23 +87,27 @@
   (:communities/community-id-input db))
 
 (defn- handle-my-request
-  [db {:keys [community-id state] :as request}]
-  {:db (if (= constants/community-request-to-join-state-pending state)
-         (assoc-in db [:communities/my-pending-requests-to-join community-id] request)
-         (update-in db [:communities/my-pending-requests-to-join] dissoc community-id))})
+  [db {:keys [community-id state deleted] :as request}]
+  (if (and (= constants/community-request-to-join-state-pending state) (false? deleted))
+    (assoc-in db [:communities/my-pending-requests-to-join community-id] request)
+    (update-in db [:communities/my-pending-requests-to-join] dissoc community-id)))
 
 (defn handle-admin-request
-  [db {:keys [id community-id] :as request}]
-  {:db (assoc-in db [:communities/requests-to-join community-id id] request)})
-
-(rf/defn handle-request-to-join
-  [{:keys [db]} r]
-  (let [my-public-key                                    (get-in db [:multiaccount :public-key])
-        {:keys [id community-id public-key] :as request} (<-request-to-join-community-rpc r)
-        my-request?                                      (= my-public-key public-key)]
-    (if my-request?
-      (handle-my-request db request)
-      (handle-admin-request db request))))
+  [db {:keys [id community-id deleted] :as request}]
+  (if (true? deleted)
+    (update-in db [:communities/requests-to-join community-id] dissoc id)
+    (assoc-in db [:communities/requests-to-join community-id id] request)))
+
+(rf/defn handle-requests-to-join
+  [{:keys [db]} requests]
+  (let [my-public-key (get-in db [:multiaccount :public-key])]
+    {:db (reduce (fn [db {:keys [public-key] :as request}]
+                   (let [my-request? (= my-public-key public-key)]
```

Comment:
Here you can use `if-let`.

Reply:
I believe we can not use `if-let` as the body would be executed only for the sender's requests. We need to process incoming requests too. The variable `my-request?` is used to check and update the appropriate key in the app-db.